src/Generator/Generators/CLI/CLIMarshal.cs

```diff
@@ -789,23 +793,26 @@ private void MarshalValueClassProperty(Property property, string marshalVar)
             if (!string.IsNullOrWhiteSpace(marshal.Context.Before))
                 Context.Before.Write(marshal.Context.Before);
 
-            Type type;
-            Class @class;
-            var isRef = property.Type.IsPointerTo(out type) &&
-                !(type.TryGetClass(out @class) && @class.IsValueType) &&
-                !type.IsPrimitiveType();
-
-            if (isRef)
+            if (!string.IsNullOrWhiteSpace(marshal.Context.Return))
```

Comment:
Why did you add this check here?

It is my understanding that the `Return` here should always provide a valid value upon invoking the marshaler. If not, then it should an error, no?

Reply:
Hi. This is for the scenario where we are marshalling an array member of a value/struct type. The array member is set and marshalled inside VisitArrayType on line 426. There is no return involved in that case, so we should not set a return here.

Basically without the if you end up with code like this:
```
    auto _marshal0 = ::UnionNestedInsideStruct();
    if (value.SzText != nullptr)
    {
        if (value.SzText->Length != 10)
        {
            throw gcnew System::InvalidOperationException("Source array size must equal destination array size.");
        }
        for (int i = 0; i < 10; i++)
            _marshal0.szText[i] = value.SzText[i];
    }
    _marshal0.szText = ;
    ((::StructWithNestedUnion*)NativePtr)->nestedUnion = _marshal0;
```

Notice the `    _marshal0.szText = ;`.